### ISSUE TITLE
Nonlinear 3D Face Morphable Model

### ISSUE THREAD

**Issue by ghuijo** [One-Day-One-Paper/One-Day-One-Paper#80]
*"3DMM 관련 논문"*

#### 도입
* 3DMM is learnt through supervision by performing dimension reduction
* To model highly variable 3D face shapes, a large amount of high-quality 3D face scans is required
* almost all models use less than 300 training scans
  --> Such a small training set is far from adequate to describe the full variability of human faces
* the texture model of 3DMM is normally built with a small number of 2D face images co-captured with 3D scans,
  under *"well-controlled conditions"*
* linear 3DMM formulation 제안
  * utilize two network decoders
    * the multi-layer perceptron (MLP) for shape
    * convolutional neural network (CNN) for texture
  * learn the fitting algorithm to nonlinear 3DMM formulated as a CNN encoder
    * encoder takes a 2D face image as input
    * generates the shape and texture parameters
    * a differentiable rendering layer to generate a reconstructed face
      by fusing the 3D face, texture, and the camera projection parameters estimated by the encoder
  *  the encoder and two decoders are learnt jointly
    to minimize the difference between the reconstructed face and the input face

#### Proposed Method
<image src="https://user-images.githubusercontent.com/67627471/153757090-ab958c30-3ccc-4ecd-b2bf-dc75e5a4759b.png" width="80%" height="80%" >

##### Conventional Linear 3DMM
* (1) 3D face is projected onto the image plane with the weak perspective projection model
  ![image](https://user-images.githubusercontent.com/67627471/153757158-9cb527c0-3b65-43da-9e9f-2459e8172fea.png)
* (2) the 2D image is rendered using texture and an illumination model

##### Nonlinear 3DMM
* learn a nonlinear 3DMM model using only large-scale in-the-wild 2D face images
* design a novel learning scheme to learn a deep 3DMM and its inference (or fitting) algorithm
* use two deep networks to decode the shape, texture parameters into the 3D facial shape and texture
* parameters are estimated by an encoder network --> essentially the fitting algorithm of our 3DMM
* objective function
  ![image](https://user-images.githubusercontent.com/67627471/153757293-e5966907-dd75-4a03-be20-7b6787688bd9.png)
* Assuming that the face mesh has the top pointing up the y axis, the projection of vS = (x, y, z) onto the UV space vT = (u, v) is computed
* α1, α2, β1, β2: constant scale and translation scalars to place the unwrapped face into the image boundaries
  ![image](https://user-images.githubusercontent.com/67627471/153757328-82181aaf-8ff0-41e2-b4a4-f0392389c8e1.png)
* In-Network Face Rendering
  ![image](https://user-images.githubusercontent.com/67627471/153757393-3f53671f-4a22-478d-a39a-49ec1c03c1ce.png)

  ![image](https://user-images.githubusercontent.com/67627471/153757405-7182e709-3c2d-4fd6-bc0d-e56f9ea3a9ee.png)

##### Network Architecture
* loss function
  ![image](https://user-images.githubusercontent.com/67627471/153757445-784a07ed-e715-440c-b67c-8857e725ed73.png)
* reconstruction loss
  ![image](https://user-images.githubusercontent.com/67627471/153757458-799fd0e9-c2af-4171-9d67-e6da3e2bdd0c.png)

#### 결과
* 3D scan reconstruction comparison (NME)
  ![image](https://user-images.githubusercontent.com/67627471/153757500-3d8415b5-9409-45d1-a8a4-5f7b5a1768d6.png)

* Face alignment performance on ALFW2000
  ![image](https://user-images.githubusercontent.com/67627471/153757513-d1aae523-fcdb-45a3-88bd-418a5af7ac11.png)

* input의 shape를 따고 여기에 맞춰 texture를 넣어 recon.
  ![image](https://user-images.githubusercontent.com/67627471/153757579-8e5510dd-1bb6-4208-8314-2e4c35ed7af3.png)
